### PR TITLE
Add check for chunking of dataset for filters

### LINKED ISSUE #212: `write_scalar` causes unknown library error
I add a simple test to my fork. It passes on my local computer but fails on many platforms, see CI results here: https://github.com/kaizhang/hdf5-rust/actions/runs/2168628529

The test code is:

```rust
#[cfg(test)]
mod tests {
    use super::*;
    use crate::test::with_tmp_file;

    #[test]
    fn test_read_write_scalar() {
        with_tmp_file::<Result<_>, _>(|file| {
            let val: f64 = 0.2;
            let dataset = file.new_dataset::<f64>().deflate(3).create("foo")?;
            dataset.write_scalar(&val)?;
            let val_back = dataset.read_scalar()?;
            assert_eq!(val, val_back);
            Ok(())
        })
        .unwrap()
    }
}
```

### PR DESCRIPTION
**Pull request by mulimoen**
The behaviour in `h5py` is to raise an error when trying to apply chunking/compression filters to dataset. This PR adds the same behaviour.

Fixes #212

### CODE REVIEW

File: hdf5/src/hl/filters.rs [hunk 1]
```diff
@@ -517,11 +517,15 @@ impl Filter {
             Ok(filters)
         })
     }
+
+    pub(crate) fn requires_chunking(&self) -> bool {
+        COMP_FILTER_IDS.contains(&self.id())
```
Comment (aldanor) [inline hdf5/src/hl/filters.rs:522]: Just wondering, that's what h5py does as well? I.e. just check a list of hardcoded well-known filter ids (because obviously there's other third party ones...)

I guess there's not much else you can do because there's no supplied filter metadata

Comment (mulimoen) [reply hdf5/src/hl/filters.rs:522]: I am not sure if a user-supplied filter needs to have filtering? The behaviour if the filter is not known is to pass the error through, worst case the user gets a confusing error

Comment (aldanor) [reply hdf5/src/hl/filters.rs:522]: I think this is pretty much the whole list of what people actually use: https://github.com/silx-kit/hdf5plugin/tree/main/src

Comment (mulimoen) [reply hdf5/src/hl/filters.rs:522]: The list is different from ours but is similar https://github.com/h5py/h5py/blob/49d7b8cde724a69a50d6cd4b218df7a6ba141633/h5py/_hl/dataset.py#L89

Comment (aldanor) [reply hdf5/src/hl/filters.rs:522]: Should we do the same (+blosc) then?

Although I'm not entirely sure all the non-compression filters they have listed _actually_ require chunking 🤔

Comment (mulimoen) [reply hdf5/src/hl/filters.rs:522]: From https://support.hdfgroup.org/HDF5/doc/H5.user/Filters.html it seems all filters build on chunked data, so the only valid `required_chunking` is the one that returns `true`

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +22,7 @@
 - Fixed a bug where errors were only silenced on the main thread.
 - Fixed a memory leak when opening datasets.
 - Avoid creating unaligned references in `H5Type` derive macro.
+- Fixed compression filter not giving an error on non-chunked datasets.
```
Comment (aldanor) [inline CHANGELOG.md:25]: Now that we've changed the check, maybe reword it as "Applying filters without chunking will now produce an explicit error."?

**Comment by aldanor** [comment]
Could you please rebase on master now that there's new stuff merged in?

File: CHANGELOG.md (hunk 2 above)
Comment (mulimoen) [reply CHANGELOG.md]: Fixed, thanks for keeping this up to date

**Review by aldanor — APPROVED** [review]
